server/hotrod/src/test/java/org/infinispan/server/hotrod/test/ByteBufTestUtil.java

```diff
@@ -0,0 +1,54 @@
+package org.infinispan.server.hotrod.test;
+
+import org.infinispan.commons.util.Util;
+import org.infinispan.server.core.transport.VInt;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.util.CharsetUtil;
+
+public final class ByteBufTestUtil {
+
+   private ByteBufTestUtil() { }
+
+   public static int readUnsignedShort(ByteBuf bf) {
+      return bf.readUnsignedShort();
+   }
+
+   public static int readUnsignedInt(ByteBuf bf) {
+      return VInt.read(bf);
+   }
+
+   public static long readUnsignedLong(ByteBuf buf) {
+      byte b = buf.readByte();
```

Comment:
Yeah, I had to leave the previous parser in place for tests :-1: 
Basically, the use of the replaying decoder makes stuff much harder. It is much more strict and keeps throwing the REPLAY signal when it would be fine since we did our checks before.